scripts/scripts.js

```diff
@@ -676,6 +687,40 @@ export const premiumArticleBanner = (customLeftoverArticles = null) => {
     </div>
   `;
 };
+
+export const validateEmail = (email) => email.match(
+  /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
+);
+
+/**
+   * Creates a class name compatible with window.store.query
+   * so page templates can also use that functionality like blocks.
+   * @param {string} Metadata string of the page that includes "(SHEET_NAME SPREADSHEET_NAME)"
+   * @return {string} String that is a valid class for window.store.query
+   */
+export const extractQueryFromTemplateMetaData = (metaDataTemplateString) => {
+  try {
+    const regex = /\((.*)\)/;
+    const textWithinBrackets = metaDataTemplateString.match(regex)[1];
+    const classesArray = textWithinBrackets.split(',').map((string) => string.trim());
+
+    const foundSpreadsheet = classesArray.find((classString) => {
+      for (const spreadsheet of window.store.spreadsheets) {
+        if (classString.endsWith(spreadsheet)) {
+          return true;
+        }
+      }
+      return false;
+    });
+
+    return foundSpreadsheet ? foundSpreadsheet.replaceAll(' ', '-') : false;
```

Comment:
You might want to reuse bits of https://github.com/adobe/helix-onedrive-support/blob/06a32ca7486fad83d38ed4e2b8507b53822c7191/src/utils.js#L36-L43 to support edge cases like multiple spaces.

Reply:
using toClassName instead. It gets the job done perfectly